TweetMate/src/main/java/com/ybsystem/tweetmate/fragments/timeline/TopicTimeline.java

```diff
@@ -0,0 +1,124 @@
+package com.ybsystem.tweetmate.fragments.timeline;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.ybsystem.tweetmate.R;
+import com.ybsystem.tweetmate.adapters.recycler.TweetRecyclerAdapter;
+import com.ybsystem.tweetmate.application.TweetMateApp;
+import com.ybsystem.tweetmate.models.entities.twitter.TwitterStatus;
+import com.ybsystem.tweetmate.utils.ExceptionUtils;
+import com.ybsystem.tweetmate.utils.ToastUtils;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.observers.DisposableObserver;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import twitter4j.Query;
+import twitter4j.QueryResult;
+import twitter4j.TwitterException;
+
+import static com.ybsystem.tweetmate.resources.ResString.*;
+
+public class TopicTimeline extends TimelineBase {
+
+    private Query mQuery;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Inflate
+        View view = inflater.inflate(R.layout.fragment_timeline, container, false);
+
+        // Init
+        mScrollLoad = true;
+
+        // Set view
+        setRecyclerView(view);
+        setRecyclerAdapter(new TweetRecyclerAdapter());
+        setUltraPullToRefresh(view);
+
+        return view;
+    }
+
+    @Override
+    protected void loadTweet(boolean isPullLoad, boolean isClickLoad) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 27 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS3776)